### PR TITLE
[Content] [Balance] Thread, yarn, and fabric audit and additions.

### PR DESCRIPTION
**Pull request by kasanryukin**
#### Summary
Content "Addition of fabric store and audit on thread and yarn recipes"

#### Purpose of change
Kevin suggested an audit of thread and yarn from an article he read, and my wife happens to semi-professionally make her own thread and yard, so I audited the existing recipes.   Turning felt directly into thread has been completely removed because you can't draft felt directly. In it's place is a new recipe to card felt back into a piece of wool_staple which can then be processed into yarn.  While in the real world, thread can be made from many different textiles, including wool, the game separates thread and yarn based on the source material and I have no immediate plan to change that. Additionally the fiber twisting proficiency has been removed from yarn recipes since there is functionally no difference between spinning thread and spinning yarn. I had both demonstrated to me from the same initial wool batting.  The time between having the spinning proficiency and not has been increased 6 fold based on the real amount of time it took me compared to my wife in making 50 inches of thread and yarn.

#### Describe the solution
Addition of overmap special Fabric Store.  Eagle-eyed players will notice it looks similar to the fur clothing shop.
Removal of felt to thread recipes.
Addition felt to wool recipe.
Proficiency adjustments related to thread and yarn

#### Describe alternatives you've considered
Leaving the fiber twisting in place,  If I hadn't thought to ask the difference between spinning thread and yarn.

#### Testing
Recipes and proficiences function as expect.  Fabric store spawns without errors, but I haven't found one naturally spawned yet. 

#### Additional context
Store
image

Recipe - No Prof
image

Recipe - Prof
image

### CODE REVIEW

**Comment by XygenSS** [comment]
could be nice to have multiple stages of spinning proficiency as to gradually improve the crafting speed.

**Comment by fairyarmadillo** [comment]
>  I have no immediate plan to change that

Probably out of scope for now, but I've been thinking we should make yarn analagous to string rather than thread, and add a wool thread item.

**Comment by kasanryukin** [comment]
> > I have no immediate plan to change that
> 
> Probably out of scope for now, but I've been thinking we should make yarn analagous to string rather than thread, and add a wool thread item.

I don't think a wool thread item is necessary per say.  A recipe to turn wool into thread sure, but unless you're going for something anachronistic, the source of the thread doesn't matter a whole lot.  Wool and linen thread were the norm in the old world until the new world was discovered and cotton taken back across the sea.  It wasn't until the industrial revolution before cotton overtook wool and linen and sources of textiles.  We have some wool thread here, but it's mostly for using in period pieces we do for the SCA, like wool thread embroidery.

**Comment by fairyarmadillo** [comment]
Wool thread would function exactly like normal thread, yes. It would get added to the group in requirements.json that includes all other thread types so that most crafts would be agnostic to what they used.

Wool thread might become more important if the player set up a loom. Wool cloth (except for felt) is made from thread, no?

**Comment by kasanryukin** [comment]
> Wool thread would function exactly like normal thread, yes. It would get added to the group in requirements.json that includes all other thread types so that most crafts would be agnostic to what they used.
> 
> Wool thread might become more important if the player set up a loom. Wool cloth (except for felt) is made from thread, no?

Basically yes.  Depending on the weight of the thread, you'd get anything from extremely soft to very rigid fabric.  Doubly so between wool, flax, and cotton.

**Comment by kasanryukin** [comment]
> Wool thread would function exactly like normal thread, yes. It would get added to the group in requirements.json that includes all other thread types so that most crafts would be agnostic to what they used.
> 
> Wool thread might become more important if the player set up a loom. Wool cloth (except for felt) is made from thread, no?

Actually this was a sound idea, so I've added wool to thread and cotton to yarn.  The source doesn't matter really since the finished product is still just thread or yarn.

**Comment by kasanryukin** [comment]
> > Actually this was a sound idea, so I've added wool to thread and cotton to yarn. The source doesn't matter really since the finished product is still just thread or yarn.
> 
> That won't work. They need to be distinct items so that they respect their materials. Wool and cotton have very different material properties in this game and it would be inappropriate to, for example, make cotton into yarn and then use that yarn to magically knit a wool sweater.
> 
> What's more, one of our traits is wool allergy. Certain characters cannot wear things made of wool. If someone with an allergy tried making yarn out of cotton hoping to get a knitted cotton outfit, they'd be sorely disappointed.
> 
> This isn't fixable without duplicating every single knitted item in the game and manually renaming them, redescing them, and making all of their recipes look for cotton yarn instead of wool.
> 
> That would be a colossal undertaking for almost no benefit, and it would add a lot of unhelpful clutter.
> 
> What you want to do is revert your last change so that cotton can't be made into yarn, and make a new item called "wool thread" which is identical to thread except for its material. That item can get added with regular thread in data/json/requirements.
> 
> Or just revert and don't bother with wool thread for now, as it's a lot of faff for little benefit.

Man I totally forgot about wool allergies, good call.

File: data/json/mapgen/s_fabricstore.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
+[
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "s_fabricstore" ],
+    "//": "Frabic Outlet",
```
Comment (Procyonae) [inline data/json/mapgen/s_fabricstore.json:6]:
```suggestion
    "//": "Fabric Outlet",
```
Pretty obvious from the id anyway but

File: data/json/mapgen/s_fabricstore.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "weight": 100,
+    "object": {
```
Comment (Procyonae) [inline data/json/mapgen/s_fabricstore.json:8]:
```suggestion
    "object": {
```
A map's "weight" field defaults to 1000 if undefined and is functional when there are multiple OMTs with the same id

File: data/json/mapgen/s_fabricstore.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        ".....0  [[   +___#......",
+        "....z#|+||i|||_St#......",
+        "....z#d |    #####......",
+        "....z#ds|LlLl#4zzz......",
+        "....z##0######z.........",
+        "........................",
+        "........................",
+        "........................"
+      ],
+      "palettes": [ "standard_building_palette", "fabric_store_pallete" ],
```
Comment (Procyonae) [inline data/json/mapgen/s_fabricstore.json:36]:
```suggestion
      "palettes": [ "standard_building_palette", "fabric_store_pallete" ],
```
There's no real benefit to defining your own palette for a single map especially when you've got unincluded terrains below, it just makes it more annoying to edit bc you have to cross reference the two you can leave it as is tho

**Review by Procyonae — COMMENTED** [review]
These aren't particularly important just pointing them out, looks good otherwise ^_^

**Pull request merged**